kasa/smart/smartchilddevice.py

```diff
@@ -107,16 +107,26 @@ async def create(
     @property
     def device_type(self) -> DeviceType:
         """Return child device type."""
-        category = self.sys_info["category"]
-        dev_type = self.CHILD_DEVICE_TYPE_MAP.get(category)
-        if dev_type is None:
-            _LOGGER.warning(
-                "Unknown child device type %s for model %s, please open issue",
-                category,
-                self.model,
-            )
-            dev_type = DeviceType.Unknown
-        return dev_type
+        if self._device_type is not DeviceType.Unknown:
+            return self._device_type
+
+        if self.sys_info and (category := self.sys_info.get("category")):
+            dev_type = self.CHILD_DEVICE_TYPE_MAP.get(category)
+            if dev_type is None:
+                _LOGGER.warning(
+                    "Unknown child device type %s for model %s, please open issue",
```

Comment:
Maybe add a test for this behavior, so that we don't accidentally break it later.

Reply:
Done

Reply:
Yes I did, have reverted that now thanks